### ISSUE TITLE
UX requirements for invitations

### ISSUE THREAD

**Issue by staltz** [ssbc/ssb-tribes2#55]
@Powersource and I just came out of a meeting with `nonlinear` where we discussed requirements for the invitation system.

The invited person should be able to see some information about the group, *without* having to replicate actual group feeds etc.

We also need to support the inviter to *revoke* the invitation. So this means that instead of sending the group key (as we do now), we have to have multiple steps:

1. Alice: Publish a DM to Bob: "grant right to invitation", containing group information
2. Bob: Publish a DM to Alice: "accept right to invitation"
3. Alice: Publish a DM to Bob: actual group addition msg containing the group symm key
4. Bob: automatically detects the DM and then automatically "joins" the group, causing reindexEncrypted to run

Or when revoking:

1. Alice: Publish a DM to Bob: "grant right to invitation", containing group information
3. Alice: revoke the "right to invitation" such that any DM Bob publishes to "accept right to invitation" will be ignored

**Comment by Powersource** [comment]
in step 1 alice shares some information about the group to bob. alternatively points bob to the door where alice has previously advertised some information about the group.

another option could be to keep doing like we're doing now, of giving the key directly to bob, and then bob can partially replicate e.g. the member tangle to figure out the current members, and maybe some `groupInfo` tangle to find out the name of the group. to make this efficient we'd need tangle replication, which André said would be hard. Maybe it doesn't need to be efficient to begin with though? Although we've said we'll want group doors either way, so maybe reusing those is easiest.

@mixmix

**Comment by mixmix** [comment]
I dont think we should do some multi-stage invite. I think this is all v2/later stuff.

Another thought - when we have deletes we can just use that for revocation. job done.

Remember revocation is likely rare too

**Comment by staltz** [comment]
I agree that starting simpler is better, but what part of "this is all v2" do you mean, @mixmix ? Like I'm okay doing invitation revocation later, but this requirement:

> The invited person should be able to see some information about the group, without having to replicate actual group feeds etc.

Seems pretty important for storage sustainability.

**Comment by Powersource** [comment]
If someone has really tight storage/bandwidth requirements they would just have to implement tangle sync? :P And can always delete after having replicated it.

**Comment by mixmix** [comment]
> The invited person should be able to see some information about the group, without having to replicate actual group feeds etc.

EASY , we attach meta-data to the invite. If it's out of date when they finally accept, it's fine.
If the inviter lied about the group overview, there is signed evidence they are a bad actor.

_It's tempting to come up with some fancy partial replication summary reduced state thing, but we just don't need to. Real world groups work based on people saying things which are true, but might only be auditable when you enter, I don't see why we need to do better than that_

**Comment by mixmix** [comment]
cc @staltz

**Comment by staltz** [comment]
> EASY , we attach meta-data to the invite. If it's out of date when they finally accept, it's fine.
> If the inviter lied about the group overview, there is signed evidence they are a bad actor.

Yes, that's fine. I brought up the lying possibility up when discussing with nonlinear, but true, we will have signed evidence that they lied.

> Real world groups work based on people saying things which are true, but might only be auditable when you enter, I don't see why we need to do better than that

Interesting

**Comment by Powersource** [comment]
as mix said,

> I dont think we should do some multi-stage invite. I think this is all v2/later stuff.

can i down-prioritize this from "Should (needed in production)"?

**Comment by staltz** [comment]
The multi-stage invite is actually one of the milestones, so it's pretty high priority (or, if you don't want to work on it now, then that milestone money won't be unlocked).

**Comment by Powersource** [comment]
Ah hmm. It just feels pretty chunky. But I guess if it'd be with so far unused money then we might be able to do it.
